packages/nox/src-thyra/NOX_PrePostOperator_RowSumScaling.C

```diff
@@ -104,9 +114,20 @@ computeScaling(const NOX::Solver::Generic& solver)
     tmp_nox_thyra_group->computeJacobian();
   }
 
+  // Returns the right scaled Jacobian, does nothing with left scaling.
   RCP< const ::Thyra::LinearOpBase< double > > jac =
     thyra_group->getScaledJacobianOperator();
 
+  // If similarity transform is enabled, left scale matrix by
+  // D_{x}^{-1} before computing row sum to preserve spectral radius
+  // bound: row sum will then be of the system: D_{x}^{-1}JD_{x}
+  if (use_similarity_transform_jacobian_) {
+    const Teuchos::RCP< const ::Thyra::ScaledLinearOpBase<double> > scaled_jac =
+      Teuchos::rcp_dynamic_cast< const ::Thyra::ScaledLinearOpBase<double> >(jac, true);
+    auto& nonconst_scaled_jac = const_cast< ::Thyra::ScaledLinearOpBase<double>&>(*scaled_jac); 
+    nonconst_scaled_jac.scaleLeft(*(thyra_group->getInvRightWeightVector()));
```

Comment:
@bartlettroscoe @rppawlo @egphill Does Thyra actually use Tpetra's new-ish threaded functions for computing row sums and left scaling?

Reply:
@mhoemmen - I just took a look - no, it does not. We will have to update the thyra-tpetra adapters. Thanks for mentioning this.

Reply:
Thanks @rppawlo !  Check out the equilibration example in `ifpack2/example`.

Reply:
Need to implement Kokkos-aware RTOps or this will turn into an interface/adapter mess.